tutorials/genomic_alignment/genomic_alignment.py

```diff
@@ -0,0 +1,444 @@
+# files generated from a the reference which allow for efficient access.
+
+
+@dataclass
+class Reference(DataClassJSONMixin):
+    """
+    Represents a reference FASTA and associated index files.
+
+    Attributes:
+        ref_name (str): Name or identifier of the reference file.
+        ref_dir (FlyteDirectory): Directory containing the reference and any index files.
+        index_name (str): Index string to pass to tools requiring it. Some tools require just the
+        ref name and assume index files are in the same dir, others require the index name.
+        indexed_with (str): Name of tool used to create the index.
+    """
+
+    ref_name: str
+    ref_dir: FlyteDirectory
+    index_name: str | None = None
+    indexed_with: str | None = None
+
+    @classmethod
+    def from_remote(cls, url: str):
+        ref_dir = Path("/tmp/reference_genome")
+        ref_dir.mkdir(exist_ok=True, parents=True)
+        ref = fetch_file(url, ref_dir)
+        return cls(ref.name, FlyteDirectory(path=str(ref_dir)))
+
+
+# Sequencing reads are the raw data generated from a sequencing experiment. In order to
+# parallelize processing of the (extremely) long strands of DNA contained in each cell,
+# we first split them up into much smaller segments. The digital representation of these
+# sequenced segments are called Reads. We define a `Reads` data class to represent a sequencing
+# reads sample via its associated FastQ files. FastQ files are simply long text files of these
+# reads with associated quality scores. The class includes attributes for the sample name and
+# paths to the read files (R1 and R2). We define a similar `from_remote` method which fetches
+# a list of URLs and constructs a list of `Reads` objects.
+
+
+@dataclass
+class Reads(DataClassJSONMixin):
+    """
+    Represents a sequencing reads sample via its associated FastQ files.
+
+    This class defines the structure for representing a sequencing sample. It includes
+    attributes for the sample name and paths to the read files (R1 and R2).
+
+    Attributes:
+        sample (str): The name or identifier of the raw sequencing sample.
+        read1 (FlyteFile): A FlyteFile object representing the path to the raw R1 read file.
+        read2 (FlyteFile): A FlyteFile object representing the path to the raw R2 read file.
+    """
+
+    sample: str
+    read1: FlyteFile | None = None
+    read2: FlyteFile | None = None
+
+    def get_read_fnames(self):
+        return (
+            f"{self.sample}_1.fastq.gz",
+            f"{self.sample}_2.fastq.gz",
+        )
+
+    def get_report_fname(self):
+        return f"{self.sample}_fastq-filter-report.json"
+
+    @classmethod
+    def from_remote(cls, urls: List[str]):
+        dl_loc = Path("/tmp/reads")
+        dl_loc.mkdir(exist_ok=True, parents=True)
+
+        samples = {}
+        for fp in [fetch_file(url, dl_loc) for url in urls]:
+            sample = fp.stem.split("_")[0]
```

Comment:
maybe slightly more pythonic?
```suggestion
        for url in urls:
            fp = fetch_file(url, dl_loc)
            sample = fp.stem.split("_")[0]
```